07_errors/patrickmarabeas/syncStore.go

```diff
@@ -13,19 +13,23 @@ func (store *SyncStore) Create(puppy Puppy) (int, error) {
 		return -1, NewError(NegativeValue)
 	}
 
+	store.Lock()
 	puppy.ID = store.uuid
 	store.Store(puppy.ID, puppy)
 	store.uuid++
+	store.Unlock()
 
 	return puppy.ID, nil
 }
 
 // Read returns the puppy matching the provided uuid.
 // An empty Puppy struct is returned if the identifier does not exist.
 func (store *SyncStore) Read(id int) (Puppy, error) {
+	store.RLock()
 	if value, ok := store.Load(id); ok {
 		return value.(Puppy), nil
 	}
+	store.RUnlock()
```

Comment:
This is a bug, because you are not unlocking the store inside the `if` statement. Instead use the `defer` pattern above and not worry about it any more.